bpmn-model/src/test/java/io/camunda/zeebe/model/bpmn/builder/BoundaryEventBuilderTest.java

```diff
@@ -0,0 +1,111 @@
+            .serviceTask("task", t -> t.zeebeJobType(JOB_TYPE))
+            .boundaryEvent(
+                "error-boundary-event",
+                b -> b.errorEventDefinition().error(ERROR_CODE).errorCodeVariable("errorCode"))
+            .endEvent()
+            .done();
+
+    final ModelElementInstance errorEventDefinition =
+        instance.getModelElementsByType(ErrorEventDefinition.class).iterator().next();
+    final ExtensionElements extensionElements =
+        (ExtensionElements)
+            errorEventDefinition.getUniqueChildElementByType(ExtensionElements.class);
+
+    assertThat(extensionElements.getChildElementsByType(ZeebeError.class))
+        .hasSize(1)
+        .extracting(ZeebeError::getErrorCodeVariable)
+        .containsExactly("errorCode");
+  }
+
+  @Test
+  void testErrorBoundaryEventErrorMessageVariableCanBeSet() {
+    final BpmnModelInstance instance =
+        Bpmn.createExecutableProcess(PROCESS_ID)
+            .startEvent()
+            .serviceTask("task", t -> t.zeebeJobType(JOB_TYPE))
+            .boundaryEvent(
+                "error-boundary-event",
+                b ->
+                    b.errorEventDefinition().error(ERROR_CODE).errorMessageVariable("errorMessage"))
+            .endEvent()
+            .done();
+
+    final ModelElementInstance errorEventDefinition =
+        instance.getModelElementsByType(ErrorEventDefinition.class).iterator().next();
+    final ExtensionElements extensionElements =
+        (ExtensionElements)
+            errorEventDefinition.getUniqueChildElementByType(ExtensionElements.class);
+
+    assertThat(extensionElements.getChildElementsByType(ZeebeError.class))
+        .hasSize(1)
+        .extracting(ZeebeError::getErrorMessageVariable)
+        .containsExactly("errorMessage");
+  }
+
+  @Test
+  void testErrorBoundaryEventErrorCodeVariableAndErrorMessageVariableCanAllBeSet() {
+    final BpmnModelInstance instance =
+        Bpmn.createExecutableProcess(PROCESS_ID)
+            .startEvent()
+            .serviceTask("task", t -> t.zeebeJobType(JOB_TYPE))
+            .boundaryEvent(
+                "error-boundary-event",
+                b ->
+                    b.errorEventDefinition()
+                        .error(ERROR_CODE)
+                        .errorCodeVariable("errorCode")
+                        .errorMessageVariable("errorMessage"))
+            .endEvent()
+            .done();
+
+    final ModelElementInstance errorEventDefinition =
+        instance.getModelElementsByType(ErrorEventDefinition.class).iterator().next();
+    final ExtensionElements extensionElements =
+        (ExtensionElements)
+            errorEventDefinition.getUniqueChildElementByType(ExtensionElements.class);
+
+    assertThat(extensionElements.getChildElementsByType(ZeebeError.class))
+        .hasSize(1)
+        .extracting(ZeebeError::getErrorCodeVariable, ZeebeError::getErrorMessageVariable)
+        .containsExactly(tuple("errorCode", "errorMessage"));
+  }
```

Comment:
👍 I like it!

❌ I also would like to see one more test case, where the boundary event has an error, but we don't set the `errorCodeVariable` nor the `errorMessageVariable`. It should verify that no ZeebeError is added as an Extension Element to the Error Event Definition.